ec2/instance.py

```diff
@@ -107,21 +173,41 @@ def terminate(**kwargs):
 
     instance_id = ctx.instance.runtime_properties['instance_id']
 
+    ctx.logger.info('(Node: {0}): Terminating instance.'.format(ctx.instance.id))
+    ctx.logger.debug('(Node: {0}): Attempting to terminate instance.'
+                     .format(ctx.instance.id))
+
     try:
         EC2().terminate_instances(instance_id)
     except EC2ResponseError:
-        raise NonRecoverableError(EC2ResponseError.body)
-
-    if _state_validation(instance_id, INSTANCE_TERMINATED, TERMINATION_TIMEOUT):
-        pass
+        ctx.logger.error("""(Node: {0}): Error.
+                         Failed to terminate instance: API returned: {1}."""
+                         .format(ctx.instance.id, EC2ResponseError.body))
+        raise NonRecoverableError('(Node: {0}): Error. Failed to terminate instance: API returned: {1}.'
+                                  .format(ctx.instance.id, EC2ResponseError.body))
+
+    ctx.logger.debug("""(Node: {0}): Attempting to verify the instance is terminated.
+                     (Instance id: {1}.)"""
+                     .format(ctx.instance.id, instance_id))
+    ctx.logger.debug('(Node: {0}): Checking State: terminated, Timeout: {1}. Check Interval: {2}.)'
+                     .format(ctx.instance.id, TERMINATION_TIMEOUT, CHECK_INTERVAL))
+
+    if _state_validation(instance_id, INSTANCE_TERMINATED, TERMINATION_TIMEOUT, CHECK_INTERVAL):
+        ctx.logger.info('(Node: {0}): Instance stopped. (Instance id: {1}).'
+                        .format(ctx.instance.id, instance_id))
     else:
-        raise RecoverableError('Instance did not terminate within specified timeout: {0}.'
-                               .format(TERMINATION_TIMEOUT))
+        ctx.logger.error('(Node: {0}): Failed to verify that the instance is terminated. (Instance id: {1}).'
+                         .format(ctx.instance.id, instance_id))
+        del ctx.instance.runtime_properties['instance_id']
+        raise NonRecoverableError('(Node: {0}): Instance did not terminate within specified timeout: {0}.'
+                                  .format(ctx.instance.id, TERMINATION_TIMEOUT))
+
 
+def _state_validation(instance_id, state, timeout_length, check_interval):
 
-def _state_validation(instance_id, state, timeout_len):
+    ctx.logger.debug('Beginning state validation: instance id: {0}, state: {1}, timeout length: {2}')
 
-    timeout = time.time() + timeout_len
+    timeout = time.time() + timeout_length
 
     while True:
         instance_state = EC2().get_all_instance_status(instance_ids=instance_id)[0]
```

Comment:
added an operation to validate the instance id refers to an actual instance and only then validate the state